docs/source/project_documentation/dev_guide/project_structure.md

```diff
@@ -0,0 +1,81 @@
+# Project Structure
+
+## Python project
+
+The backbone of AaC is the python project which implements the underlying
+functionality for AaC, as a whole.
+
+### cli
+
+Since AaC is built as a CLI-first tool in order to maintain a minimal footprint,
+enable integration into CI/CD pipelines, etc, we require functionality for
+allowing users to interact with the application via a command line. The `cli`
+package provides this functionality for users including an internal
+representation of AaC commands, translation between our internal structures and
+the [click][1] CLI framework which we use for CLI interaction, etc.
+
+Additionally, in the `cli.builtin_commands` package you will find AaC plugins
+that provide functionality without which AaC could not operate or could not
+operate effectively.
+
+### io
+
+As the name suggests, the `io` package provides functionality related to I/O
+operations including parsing of AaC files.
+
+### lang
+
+The `lang` package provides the main programmatic interface for interacting with
+AaC structures via the "Active Context" (which is an instance of a
+`LanguageContext`). It also provides various utilities to enable working with
+`Definition`s.
+
+### persistence
+
+At one point in time it was desirable to provide a level of persistence for the
+`ActiveContext` so the `persistence` package was created to handle that.
+Currently, however, it is unused and liable to be removed in a future version.
+
+### plugins
+
+At the heart of AaC's design philosophy is the desire to keep the core tool as
+minimal as possible and allow for enhancing AaC functionality via plugins. To
+balance this desire with the idea of providing enough additional functionality
+to allow AaC to be useful "out-of-the-box", we use the `plugins` package.
+
+The `plugins` package is similar to the `cli.builtin_commands` package except
+for the fact that every plugin in the `plugins.first_party` package and every
+validator in the `plugins.validators` (save for one - the `required_fields`
+validator) could be deployed separately from the core AaC and the tool would
+still work.
+
+### serialization
+
+We understand that not all AaC users will enjoy working from a terminal so we've
```

Comment:
```suggestion
We understand that not all AaC users will enjoy working from a terminal. So, we've
```

Reply:
Technically I believe it should just be a comma before 'so'.